fcrepo-persistence-ocfl/src/test/java/org/fcrepo/persistence/ocfl/impl/UpdateNonRdfSourcePersisterTest.java

```diff
@@ -0,0 +1,190 @@
+import org.fcrepo.persistence.ocfl.api.OCFLObjectSession;
+import org.fcrepo.persistence.ocfl.api.OCFLObjectSessionFactory;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+/**
+ * @author whikloj
+ * @since 6.0.0
+ */
+@RunWith(MockitoJUnitRunner.Silent.class)
+public class UpdateNonRdfSourcePersisterTest {
+
+    @Mock
+    private NonRdfSourceOperation nonRdfSourceOperation;
+
+    @Mock
+    private OCFLObjectSession session;
+
+    @Mock
+    private FedoraOCFLMapping mapping;
+
+    @Mock
+    private FedoraToOCFLObjectIndex index;
+
+    @Mock
+    private OCFLObjectSessionFactory objectSessionFactory;
+
+    @Mock
+    private WriteOutcome writeOutcome;
+
+    @Captor
+    private ArgumentCaptor<InputStream> userContentCaptor;
+
+    @Captor
+    private ArgumentCaptor<InputStream> headersIsCaptor;
+
+    @Mock
+    private OCFLPersistentStorageSession psSession;
+
+    private static final String RESOURCE_ID = "info:fedora/parent/child";
+
+    private static final String ROOT_RESOURCE_ID = "info:fedora/parent";
+
+    private static final String USER_PRINCIPAL = "fedoraUser";
+
+    private static final String CONTENT_BODY = "this is some example content";
+
+    private static final Long LOCAL_CONTENT_SIZE = Long.valueOf(CONTENT_BODY.length());
+
+    private static final String EXTERNAL_URL = "http://example.com/file.txt";
+
+    private static final String EXTERNAL_HANDLING = "proxy";
+
+    private static final Long EXTERNAL_CONTENT_SIZE = 526632l;
+
+    private UpdateNonRdfSourcePersister persister;
+
+    @Before
+    public void setUp() throws Exception {
+        when(mapping.getOcflObjectId()).thenReturn("object-id");
+        when(mapping.getRootObjectIdentifier()).thenReturn(ROOT_RESOURCE_ID);
+
+        when(session.write(anyString(), any(InputStream.class))).thenReturn(writeOutcome);
+
+        nonRdfSourceOperation = mock(NonRdfSourceOperation.class, withSettings().extraInterfaces(
+                CreateResourceOperation.class));
+        when(nonRdfSourceOperation.getUserPrincipal()).thenReturn(USER_PRINCIPAL);
+        when(nonRdfSourceOperation.getResourceId()).thenReturn(RESOURCE_ID);
+        when(nonRdfSourceOperation.getContentSize()).thenReturn(null);
+
+        when(writeOutcome.getContentSize()).thenReturn(LOCAL_CONTENT_SIZE);
+        when(psSession.findOrCreateSession(anyString())).thenReturn(session);
+        when(index.getMapping(anyString())).thenReturn(mapping);
+
+        persister = new UpdateNonRdfSourcePersister(objectSessionFactory, index);
+
+        when(psSession.findOrCreateSession(anyString())).thenReturn(session);
+        when(index.getMapping(anyString())).thenReturn(mapping);
+
+    }
+
+    @Test
+    public void testNonRdfExistingResource() throws Exception {
```

Comment:
test for updating a resource that doesn't exist

Reply:
This persister should never be called when the resource does not exist. That case will be handled by CreateNonRDFSourcePersister.

Reply:
I believe we'll be checking at a higher level for this.